osutil/epoll/epoll.go

```diff
@@ -172,29 +191,36 @@ func (e *Epoll) WaitTimeout(duration time.Duration) ([]Event, error) {
 			bufLen = 1
 		}
 		sysEvents = make([]unix.EpollEvent, bufLen)
-		startTs := time.Now()
 		n, err = unixEpollWait(e.fd, sysEvents, msec)
```

Comment:
there's a small race between the IsClosed check and here,  should we consider EBADF errors also as ErrEpollClosed ?

so for the other main places doing syscalls on e.fd

Reply:
Yes, that's what I had in mind. If `Close()` closes `e.fd`, then that may trigger the syscall to return `EBADF`. However, we close `e.closed` before closing `e.fd`, so by the time the syscall would fail as a result, the check of `IsClosed()` that immediately follows will return `true` and the function will send `ErrEpollClosed` and return.

My logic is:
1. Closing `e.fd` is the only reason we should see the syscall return `EBADF`, so we generally want to treat this as `ErrEpollClosed`
2. After the epoll has been closed, we don't care if there happened to be an unrelated `EBADF` somehow, we should just return `ErrEpollClosed` anyway because that's the more pressing issue --- the epoll instance should not be used anymore

I've adjusted the other syscalls on `e.fd` to also make a check for `IsClosed()` after they return. Does this sound good to you?